polls.py

```diff
@@ -81,100 +89,228 @@ def get_role(self, emoji_unicode: str) -> (discord.Role, List[Tuple[str, discord
-                await member.add_roles(role)
-                await member.add_roles(self.general_role)
+        # get context objects back
+        channel = bot.get_channel(eval(lines.pop(0)))
+        self.message = await channel.fetch_message(eval(lines.pop(0)))
+        self.general_role = channel.guild.get_role(eval(lines.pop(0)))
+        self.author = channel.guild.get_member(eval(lines.pop(0)))
 
-            except Exception as eEe:
-                print(f"Reaction watch loop error: {eEe}")
+        for line in lines:  # add roles back to lists
+            self.emoji_role_paired_list.append([line[0], channel.guild.get_role(eval(line[1:]))])
+            self.roles.append(channel.guild.get_role(eval(line[1:])))
+        return self
 
-    async def add_reactions(self):
+    @staticmethod
+    async def loadall(bot) -> List:
         """
-        Adds the appropriate reactions to the poll message, and the cancel emoji to the emoji list, also names the
-        general reaction role (role you get from reacting to anything) after the title
+        This iterates through all of the cached polls in `cache/polls`, un-pickles them, and adds
+        back the context attributes. Then it returns the list of polls to be run
+
+        :param bot: Client connection to discord
+        :type bot: Object
+        :return: A list of the cached polls
+        :rtype: List[Poll]
         """
-        for emoji in self.emoji_list:
-            await self.message.add_reaction(emoji)
-        await self.message.add_reaction('❌')
-        await self.message.add_reaction('⛔')
-        self.emoji_list.append('❌')
-        self.emoji_list.append('⛔')
+        try:
+            await bot.wait_until_ready()
+            poll_list = []
+            pickle_file_list = sorted(os.listdir('cache/polls'))[::2]
+            potato_file_list = sorted(os.listdir('cache/polls'))[1::2]
+
+            for num, file in enumerate(pickle_file_list):
+                with open('cache/polls/' + file, 'rb') as read_file:
+                    poll_list.append(pickle.load(read_file))
+
+                with open('cache/polls/' + potato_file_list[num], 'r') as read_file2:
+                    lines = read_file2.readlines()
+
+                channel: discord.TextChannel = bot.get_channel(eval(lines.pop(0)))
+                poll_list[num].message = await channel.fetch_message(eval(lines.pop(0)))
+                poll_list[num].general_role = channel.guild.get_role(eval(lines.pop(0)))
+                poll_list[num].author = channel.guild.get_member(eval(lines.pop(0)))
+
+                for line in lines:  # add roles back to lists
+                    poll_list[num].emoji_role_paired_list\
+                        .append([line[0], channel.guild.get_role(eval(line[1:]))])
+
+                    poll_list[num].roles.append(channel.guild.get_role(eval(line[1:])))
+            return poll_list
+        except Exception as ee:
```

Comment:
Whenever there is just `except Exception as e` followed by `print(e)` I'm not really catching the exception, because discord.py does that for us. I just want to print the exception, because all discord.py does is ignore it (except for `SystemExit`).